lib/internal/repl.js

```diff
@@ -164,13 +164,22 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
       }
     }
 
-    fs.open(historyPath, 'w', onhandle);
+    fs.open(historyPath, 'r+', onhandle);
   }
 
   function onhandle(err, hnd) {
     if (err) {
       return ready(err);
     }
+    fs.ftruncate(hnd, 0, (err) => {
```

Comment:
From what I am gathering, the mode change no longer cause overwrites (appends instead) and so we need to "flush" the file?

[From the docs](https://nodejs.org/dist/latest-v7.x/docs/api/fs.html#fs_fs_open_path_flags_mode_callback) though, it sounds like it does writes and not appends, so this shouldn't be necessary?

> 'r+' - Open file for reading and writing. An exception occurs if the file does not exist.